File: Taller RabbitMQ V2/publisher_python/publisher.py
import pika
import time

# Configura el intervalo de envío.
interval = 5

# Conexión a RabbitMQ
connection = pika.BlockingConnection(pika.ConnectionParameters('rabbitmq'))
channel = connection.channel()

# Declarar la cola
queue = 'hello'
channel.queue_declare(queue=queue, durable=False)

# Mensaje a enviar
msg = 'Mensaje Enviado desde Python'

# Enviar el mensaje cada cierto intervalo
try:
    while True:
        channel.basic_publish(exchange='',
                              routing_key=queue,
                              body=msg)
        print(f"Publisher Python: {msg}")
        time.sleep(interval)
except KeyboardInterrupt:
    print("Publisher Python detenido.")
finally:
    connection.close()

File: Taller RabbitMQ/publisher_node/publisher.js
const amqp = require('amqplib/callback_api');

// Configura el intervalo de envio.
const interval = 5000;

amqp.connect('amqp://rabbitmq', function(error0, connection) {
  if (error0) {
    throw error0;
  }

  connection.createChannel(function(error1, channel) {
    if (error1) {
      throw error1;
    }

    const queue = 'cola1';
    const msg = 'Mensaje Enviado desde Node';

    channel.assertQueue(queue, {
      durable: false
    });

    // Enviar el mensaje cada cierto intervalo
    setInterval(() => {
      channel.sendToQueue(queue, Buffer.from(msg));
      console.log("Publisher Node: %s", msg);
    }, interval);
  });
});
